docs/en/introduction.md

```diff
@@ -0,0 +1,99 @@
+
+Its core design concept is:
+
+> The UI is a re-description of data interaction and continues to respond to modifications of data.
+
+We emphasize "re" here because we think the API is the first description of data interaction, and building UI with Ribir requires only the data API.
+
+## Why choose Ribir?
+
+### Non-intrusive programming model
+
+Ribir only interacts with the API of your data and does not require any pre-design of your data for the user interface: no additional states, no additional notification mechanisms, no inheritance of any base classes or any other pre-constraints. It doesn't break the logic and structure of your existing data, nor does it inject any additional objects. When developing the core part of an application, you can focus on designing the data, logic, and API of the application without thinking about the UI at all.
+
+The UI directly operates data, and data modifications directly drive UI updates, without any intermediate layers and concepts.
+
+### Consistent experience across multiple platforms, and easy to expand to new platforms
+
+Ribir can be used to develop desktop, mobile, web and server-side rendering applications. It generates efficient binary code or WASM programs without relying on any runtime environment. It outputs a very simple, platform-independent drawing result, allowing you to choose to be rendered entirely by the GPU or CPU. You can even easily implement your own rendering backend to expand to uncovered platforms.
+
+### Declarative syntax that is easy to interact with Rust
+
+Ribir provides a declarative syntax that is easy to interact with Rust. It is not a new language, but a set of Rust macros. Therefore, it can interact well with Rust, making your code both a clear view description and a powerful logical expression, without any environment and tool dependencies.
+
+### Point-to-point view update strategy
+
+Ribir will map a view tree based on your description of the data, and the view will be updated in response to data modifications - this update does not rebuild the entire view, but updates the parts of the view that depend on the modified data point-to-point.
+
+The update logic is determined at compile time, and there is no general diff or patch algorithm to execute at runtime.
+
+Ribir uses a Pipe stream to declare the part of the view that needs to change continuously. It can be a property of a widget, a specific widget, or a complete subtree.
+
+### "Pay-as-you-go" design principle
+
+Because it has to face various complex real-life scenarios, a general GUI framework often has complex designs and rich capabilities, so it is difficult to be lightweight. The way Ribir balances this problem is to provide enough capabilities to ensure development efficiency, and requiring that all capabilities only need to be understood and have overhead when they are used. A few examples:
+
```

Comment:
Because it has to face various complex real-life scenarios, a general GUI framework often has complex designs and rich capabilities, so it is difficult to be lightweight.
->
Due to the need to handle multiple complex real-life situations, a general GUI framework tends to have complex designs and extensive features. As a result, it is difficult to keep it lightweight.